tivars/var.py

```diff
@@ -111,137 +61,163 @@ def checksum(self) -> int:
+        dump += self.data_length_bytes
+        dump += self.data
+        dump += self.checksum_bytes
 
-    def dump(self) -> bytes:
-        return self.header.dump() + self.entry.dump() + self.checksum.to_bytes(2, 'little')
+        return dump
 
-    def dumps(self) -> str:
-        pass
+    def load_bytes(self, data: bytes | Iterator[bytes]):
+        data = iter(data)
 
-    def load(self, file: BinaryIO):
-        self.signature = file.read(8)
-        self.export = file.read(3)
-        self.comment = file.read(42).decode('utf8')
-        entry_length = int.from_bytes(file.read(2), 'little')
+        self.load_buffer(TIVar.signature, data)
+        match self.signature:
+            case TI_82.signature:
+                model = TI_82
+            case TI_83.signature:
+                model = TI_83
+            case TI_84p.signature:
+                model = TI_84p
+            case _:
+                raise warnings.warn(f"The var signature is not recognized ({self.signature}).",
+                                    BytesWarning)
 
-        self.meta_length = int.from_bytes(file.read(2), 'little')
-        data_length = int.from_bytes(file.read(2), 'little')
+        if self.model is None:
+            self.model = model
 
-        type_id = file.read(1)
-        if type_id != self.type_id:
-            raise TypeError("The var type is incorrect. Use TIVar.infer if you don't know the type.")
+        self.load_buffer(TIVar.export, data)
+        self.load_buffer(TIVar.comment, data)
+        entry_length = int.from_bytes([next(data), next(data)], 'little')
 
-        self.name = file.read(8).decode('utf8')
+        self.load_buffer(TIVar.meta_length, data)
+        data_length = int.from_bytes([next(data), next(data)], 'little')
+
+        type_id = bytes([next(data)])
+        if self.type_id is None:
+            try:
+                self.__class__ = TIVar.type_ids[type_id]
+
+            except KeyError:
+                raise TypeError(f"Type id 0x{type_id:x} is not recognized.")
+
+        elif type_id != self.type_id:
+            raise TypeError("The var type is incorrect. Use a TIVar instance if you don't know the type.")
+
+        self.load_buffer(TIVar.name, data)
 
         if self.meta_length == TIVar.flash_meta_length:
-            self.version = file.read(1)
-            self.archived = file.read(1) != b'\x00'
+            if self.model is not None and not self.model.has(TIFeature.FLASH):
+                warnings.warn(f"Calculator model does ")
+
+            self.load_buffer(TIVar.version, data)
+            self.load_buffer(TIVar.archived, data)
 
         elif self.meta_length != TIVar.base_meta_length:
             warnings.warn(f"The var entry meta length has an unexpected value ({self.meta_length}).",
                           BytesWarning)
-            self.corrupt = True
 
-        data_length2 = int.from_bytes(file.read(2), 'little')
+        data_length2 = int.from_bytes([next(data), next(data)], 'little')
```

Comment:
wasn't this one of the things we hated about the cpp implementation

Reply:
`data_length2` isn't actually stored anywhere; its just used for file verification.